beacon_node/eth1/src/deposit_cache.rs

```diff
@@ -203,6 +205,35 @@ impl DepositCache {
             Ok((tree.root(), deposits))
         }
     }
+
+    /// Gets the deposit count at block height = block_number.
+    ///
+    /// Fetches the `DepositLog` that was emitted at or just before `block_number`
+    /// and returns the deposit count as `index + 1`.
+    ///
+    /// Returns 0 if no logs are present.
+    /// Note: This function assumes that `block_number` > `deposit_contract_deploy_block`
```

Comment:
Can't think of a good way to handle the "before deposit contract deployed" case.

Reply:
I think we will have to take note of when the deposit contract was deployed and return an error if we search prior to this. This scenario should never happen, only if a testnet has been badly configured.